lib/em-hiredis/req_resp_connection.rb

```diff
@@ -0,0 +1,77 @@
+
+module EventMachine::Hiredis
+  module ReqRespConnection
+    include EventMachine::Hiredis::EventEmitter
+
+    def initialize
+      super
+      @response_queue = []
+      @connected = false
+    end
+
+    def send_command(df, command, args)
+      @response_queue.push(df)
+      puts "send #{command} #{args}"
+      send_data(marshal(command, *args))
```

Comment:
We don't call this method before the connection is established (very important, lost of tests), but I *believe* eventmachine would queue the data and send it if the connection establishes.